main.py

```diff
@@ -0,0 +1,86 @@
+from fastapi import FastAPI, Depends, HTTPException
+from fastapi_pagination import add_pagination
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+@app.get("/")
+def root() -> dict:
+    return {"message": "Hello world"}
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(db: Session = Depends(get_db)):
+    return crud.get_all_authors(db=db)
+
+
+@app.get("/authors/{authors_id}/", response_model=schemas.Author)
+def read_single_author(authors_id: int, db: Session = Depends(get_db)):
+    db_author = crud.get_author_by_id(db=db, authors_id=authors_id)
+
+    if db_author is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Author not found"
+        )
+
+    return db_author
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorCreate,
+    db: Session = Depends(get_db),
+):
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+
+    if db_author:
+        raise HTTPException(
+            status_code=400,
+            detail="Such name for author already exist"
+        )
+
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/book/", response_model=list[schemas.Book])
+def read_book(
+    author_id: int | None = None,
+    db: Session = Depends(get_db)
+):
+    return crud.get_book_list(
+        db=db, author_id=author_id
+    )
+
+@app.get("/book/{book_id}/", response_model=schemas.Book)
+def read_single_book(book_id: int, db: Session = Depends(get_db)):
+    db_book = crud.get_book(db=db, book_id=book_id)
+
+    if db_book is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Book not found"
+        )
+
+    return db_book
+
+
+@app.post("/book/", response_model=schemas.Book)
+def create_book(book: schemas.BookCreate, db: Session = Depends(get_db)):
+    return crud.create_book(db=db, book=book)
+
+
+add_pagination(app)
```

Comment:
this task is rejected anyway due to redundant `root()` func and blank line, so can you please provide screenshot of paginated response from your endpoint? Just wanna look at it, cuz I've never used this lib)

Reply:
Refactored that thing to more usual limit-offset (now it's more readeble)

Reply:
![image](https://github.com/mate-academy/py-fastapi-library-management-api/assets/92045507/9e28b0c8-b8b8-4de9-abe3-4d1ca100d94e)